libpyvinyl/Parameters/Parameter.py

```diff
@@ -8,126 +8,168 @@ class Parameter(AbstractBaseClass):
     """
     Description of a single parameter
     """
+
     def __init__(self, name, unit=None, comment=None):
         """
         Creates parameter with given name, optionally unit and comment
         """
         self.name = name
         self.unit = unit
         self.comment = comment
-        self.value = None
-        self.legal_intervals = []
-        self.illegal_intervals = []
+        self._value = None
+        self.intervals = []
+        self._intervals_are_legal = None
         self.options = []
+        self._options_are_legal = None
 
     @classmethod
     def from_dict(cls, param_dict):
-        param = cls(param_dict['name'], param_dict['unit'],
-                    param_dict['comment'])
+        param = cls(param_dict["name"], param_dict["unit"], param_dict["comment"])
         for key in param_dict:
             param.__dict__[key] = param_dict[key]
         return param
 
-    def add_legal_interval(self, min_value, max_value):
+    def add_interval(self, min_value, max_value, intervals_are_legal):
         """
-        Sets a legal interval for this parameter, None for infinite
+        Sets an interval for this parameter, None for infinite
         """
-        if min_value is None:
-            min_value = -math.inf
-        if max_value is None:
-            max_value = math.inf
-
-        self.legal_intervals.append([min_value, max_value])
+        if self._intervals_are_legal is None:
+            self._intervals_are_legal = intervals_are_legal
+        else:
+            if self._intervals_are_legal != intervals_are_legal:
+                print("WARNING: All intervals should be either legal or illegal.")
+                print(
+                    "         Interval: ["
+                    + str(min_value)
+                    + ":"
+                    + str(max_value)
+                    + "] is declared differently w.r.t. to the previous intervals"
+                )
```

Comment:
The format I got from `black` is respecting the "line break before binary operator", since all `+` are on a new line.
So I don't really get why there is this suggestion from flake8.

To be honest the formatting I get seems to be more readable. I will try `yapf` to understand it a bit better and then we can decide what to use.

Reply:
It seems an obsolete warning. According to this page (https://www.flake8rules.com/rules/W503.html)

> The tool will soon be updated to recommend the opposite: line breaks should occur before the binary operator because it keeps all operators aligned.

In that sense, maybe `black` is the most updated one at this moment.

Reply:
Ok, I will keep it like that for the moment.

Reply:
I like the black formatting in this case.